pkg/datapath/iptables/iptables.go

```diff
@@ -1217,7 +1219,77 @@ func (m *IptablesManager) installMasqueradeRules(prog iptablesInterface, ifName,
 		}
 	}
 
-	// Masquerade all egress traffic leaving the node
+	// Masquerade egress traffic leaving the node based on source routing
+	//
+	// If this option is enabled, then it takes precedence over the catch-all
+	// MASQUERADE further below.
+	if option.Config.EnableMasqueradeRouteSource {
+		devices := option.Config.GetDevices()
+		if option.Config.EgressMasqueradeInterfaces != "" {
```

Comment:
Maybe we could change the `EgressMasqueradeInterfaces` flag type to `StringSlice`? Then the split below would not be needed.

Reply:
Ok, done.

Reply:
Tobias gave some hints how to work-around this by keeping the old EnableMasqueradeRouteSource alive as string.